src/discord-cluster-manager/cogs/verify_run_cog.py

```diff
@@ -0,0 +1,103 @@
+import discord
+from discord import app_commands
+from discord.ext import commands
+import re
+from utils import setup_logging
+
+logger = setup_logging()
+
+class VerifyRunCog(commands.Cog):
+    """
+    A Discord cog for verifying the success of trainingruns.
+
+    This cog provides functionality to verify that either a GitHub Actions or
+    Modal run completed successfully by checking for specific message patterns
+    in a Discord thread. It supports verification of two types of runs:
+    1. GitHub Actions runs - Identified by "GitHub Action triggered!" message
```

Comment:
So this seems to test a trigger but we've had issues where this wouldn't have helped like for example when we had timeout issues with both the NVIDIA and AMD runner